libtfhe/core/arithmetic/polynomial_torus.cpp

```diff
@@ -333,3 +333,234 @@ void TorusPolynomial<TORUS>::SubMulRKaratsuba(
     delete[] R;
     delete[] buf;
 }
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+// new functions
+
+
+
+template<typename TORUS, typename INT_TYPE>
```

Comment:
Add as comment that it multiplies two pre-allocated polynomials of degree N, writes the result as a degree 2N-1
and that result and poly2 must not overlap